scc-big-data.ipynb

```diff
@@ -0,0 +1,1883 @@
+    "\n",
+    "**YOUR TASK:**\n",
+    "\n",
+    "  1. Sign up for a U.S. Census Bureau API Key at their [Request a Key website](https://api.census.gov/data/key_signup.html). You can list the University of Colorado as your organization.\n",
+    "  2. Locate the Median Income in the [list of Census variables](https://api.census.gov/data/2019/acs/acs5/variables.html)\n",
+    "  3. Download the 2021 Median Income for each Census Tract, making sure to **cache your download**\n",
+    "  \n",
+    "> NOTE: The Census API will only let you download 50 tracts at once. Can you figure out how to use a loop to ask for 50 tracts at a time?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c9ba4221-076c-4c6a-bfef-96c2901e9e5b",
+   "metadata": {
+    "nbgrader": {
+     "cell_type": "code",
+     "checksum": "9505e2598968c0493f6dff974fa25204",
+     "grade": false,
+     "grade_id": "census-ans",
+     "locked": false,
+     "schema_version": 3,
+     "solution": true,
+     "task": false
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "#  Download the Census data for the state of Illinois for the year 2021\n",
+    "\n",
+    "census_data_path = os.path.join(proj_dir, 'census_data', 'il_census.csv')\n",
+    "if not os.path.exists(census_data_path):\n",
+    "    api_key = getpass.getpass('U.S. Census API Key')\n",
+    "    c = Census(api_key)\n",
+    "    il_census = c.acs5.state_county_tract(fields = ('NAME', 'B06011_001E', 'B19301_001E', 'B01003_001E', 'B25035_001E'),\n",
+    "                                      state_fips = states.IL.fips,\n",
+    "                                      county_fips = \"*\",\n",
+    "                                      tract = \"*\",\n",
+    "                                      year = 2021) \n",
+    "    il_df = pd.DataFrame(il_census)\n",
+    "    print(\"Column data types for il_df data:\\n{}\".format(il_df.dtypes))\n",
+    "    il_df.to_csv(census_data_path, index=False, header=False)\n",
+    "    \n",
+    "# Explicitly set data types to object to preserve leading zeros\n",
+    "column_names = (['NAME', 'median_income', 'per_capita_income',\n",
+    "                     'population', 'median_year_built','state', 'county', 'tract', 'GEOID'])\n",
+    "column_dtypes = ({'NAME': object, 'median_income': float, 'per_capita_income': float,\n",
+    "                      'population': float, 'median_year_built': int, 'state': object,\n",
+    "                      'county': object, 'tract': object, 'GEOID': object})\n",
+    "    \n",
+    "# Read the CSV file into a DataFrame\n",
+    "il_census_csv = pd.read_csv(census_data_path, names=column_names, dtype=column_dtypes)\n",
+    "  \n",
+    "# Show the dataframe\n",
+    "print(il_census_csv.tail(10))\n",
+    "print('Shape: ', il_census_csv.shape)\n",
+    "#  Confirm that the data is in the same format after changing it to .csv file\n",
```

Comment:
It's cool that you have a built-in quick test like we used to have the previous semester to check our progress.